api/users/serializers.py

```diff
@@ -50,3 +68,64 @@ class ContributorSerializer(UserSerializer):
     filterable_fields = frozenset.union(UserSerializer.filterable_fields, local_filterable)
 
     bibliographic = ser.BooleanField(help_text='Whether the user will be included in citations for this node or not')
+
+
+class ApiOAuth2ApplicationSerializer(JSONAPISerializer):
+    """Serialize data about a registered OAuth2 application"""
+    client_id = ser.CharField(help_text="The client ID for this application (automatically generated)",
+                              read_only=True)
+
+    client_secret = ser.CharField(help_text="The client secret for this application (automatically generated)",
```

Comment:
May want to lengthen the client secret slightly here per rfc6749. 40 characters would be a good lower bound. (2^160 possible combinations). This is 32 chars, and 2 of the fields in uuid are less random.